src/flows/pipes/QueryBuilder/context.tsx

```diff
@@ -363,21 +376,35 @@ export const QueryBuilderProvider: FC = ({children}) => {
       ? EXTENDED_TAG_LIMIT
       : DEFAULT_TAG_LIMIT
 
-    // TODO: Use the `v1.tagValues` function from the Flux standard library once
-    // this issue is resolved: https://github.com/influxdata/flux/issues/1071
-    query(
-      `${_source}
-              |> range(${formatTimeRangeArguments(range)})
-              |> filter(fn: (r) => ${tagString})
-              |> keep(columns: ["${cards[idx].keys.selected[0]}"])
-              |> group()
-              |> distinct(column: "${
-                cards[idx].keys.selected[0]
-              }")${searchString}
-              |> limit(n: ${limit})
-              |> sort()`,
-      scope
-    )
+    let queryText = `${_source}
+    |> range(${formatTimeRangeArguments(range)})
+    |> filter(fn: (r) => ${tagString})
+    |> keep(columns: ["${cards[idx].keys.selected[0]}"])
+    |> group()
+    |> distinct(column: "${cards[idx].keys.selected[0]}")${searchString}
+    |> limit(n: ${limit})
+    |> sort()`
+
+    if (
+      data.buckets[0].type !== 'sample' &&
+      isFlagEnabled('MeasurementFieldsNoTime')
+    ) {
+      _source = `import "regexp"
+      import "influxdata/influxdb/schema"`
+      queryText = `${_source}
+  schema.tagValues(
+    bucket: "${data.buckets[0].name}",
+    tag: "${cards[idx].keys.selected[0]}",
+    predicate: (r) => ${tagString},
+    start: ${CACHING_REQUIRED_START_DATE},
+    stop: ${CACHING_REQUIRED_END_DATE},
+  )${searchString}
+  |> group()
+  |> sort()
+    |> limit(n: ${limit})`
```

Comment:
I see that this reverses the order of `sort` and `limit` from the old query. This version here is consistent with the query for getting tag keys above, and it makes sense to me. Just want to call it out in case it wasn't intended.

Reply:
Good catch. I think it makes more sense to have `limit` and then `sort`. So, I have made sure that it's the same in these 4 places.